code/game/machinery/shieldgen.dm

```diff
@@ -553,3 +553,644 @@
+	if (stored_strength < 5)
+		recovering = TRUE
+		deactivate_shields()
+		stored_strength = 0
+		update_icon_state()
+		return
+
+/obj/machinery/modular_shield_gen/process(seconds_per_tick)
+	stored_strength = min((stored_strength + (current_regeneration * seconds_per_tick)),max_strength)
+	if(stored_strength == max_strength)
+		if (recovering)
+			recovering = FALSE
+			calculate_regeneration()
+			update_icon_state()
+		STOP_PROCESSING(SSobj, src) //we dont care about continuing to update the alpha, we want to show history of damage to show its unstable
+	if (active)
+		var/random_num = rand(1,deployed_shields.len)
+		var/obj/structure/emergency_shield/modular/random_shield = deployed_shields[random_num]
+		random_shield.alpha = max(255 * (stored_strength/max_strength), 40)
+
+
+
+
+//Start of other machines
+/obj/machinery/modular_shield/module //The general code used for machines that want to connect to the network
+
+	name = "Modular Shield Debugger" //Filler name and sprite for testing
+	desc = "This is filler for testing you shouldn`t see this"
+	icon = 'icons/mecha/mech_bay.dmi'
+	icon_state = "recharge_port"
+	density = TRUE
+	dir = SOUTH
+
+	var/obj/machinery/modular_shield_gen/shield_generator
+
+	var/obj/machinery/modular_shield/module/node/connected_node
+
+	var/turf/connected_turf
+
+/obj/machinery/modular_shield/module/Initialize(mapload)
+	. = ..()
+
+	connected_turf = get_step(loc, dir)
+
+/obj/machinery/modular_shield/module/Destroy()
+	. = ..()
+	if(shield_generator)
+		shield_generator.connected_modules -= (src)
+		shield_generator.calculate_boost()
+	if(connected_node)
+		connected_node.connected_through_us -= (src)
+	return ..()
+
+/obj/machinery/modular_shield/module/attackby(obj/item/I, mob/user, params)
+
+	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		panel_open = !(panel_open)
+		I.play_tool_sound(src, 50)
+		update_icon_state()
+		if(panel_open)
+			to_chat(user, span_notice("You open the maintenance hatch of [src]."))
+			return TRUE
+		to_chat(user, span_notice("You close the maintenance hatch of [src]."))
+		return TRUE
+
+//rather than automatically checking for connections its probably alot less expensive to just make the players manually multi tool sync each part
+	if(I.tool_behaviour == TOOL_MULTITOOL)
+		try_connect(user)
+		return
+
+	if(default_change_direction_wrench(user, I))
+		if(shield_generator)
+			shield_generator.connected_modules -= (src)
+			shield_generator.calculate_boost()
+			shield_generator = null
+			update_icon_state()
+		if(connected_node)
+			connected_node.connected_through_us -= (src)
+			connected_node = null
+		connected_turf = get_step(loc, dir)
+		return
+
+	if(default_deconstruction_crowbar(I))
+		return
+	return ..()
+
+/obj/machinery/modular_shield/module/setDir(new_dir)
+	. = ..()
+	connected_turf = get_step(loc, dir)
+
+/obj/machinery/modular_shield/module/proc/try_connect(user)
+
+	if(shield_generator || connected_node)
+		balloon_alert(user, "already connected to something")
```

Comment:
```suggestion
		balloon_alert(user, "already connected to something!")
```

error states should have exclamation points.

Reply:
bump

Reply:
commited manually